lib/spack/spack/cmd/versions.py

```diff
@@ -20,9 +20,12 @@
 
 
 def setup_parser(subparser):
-    subparser.add_argument('-s', '--safe-only', action='store_true',
-                           help='only list safe versions of the package')
-    subparser.add_argument('-l', '--latest', action='store_true',
+    output = subparser.add_mutually_exclusive_group()
+    output.add_argument('-s', '--safe-only', action='store_true',
+                        help='only list safe versions of the package')
+    output.add_argument('-r', '--remote-only', action='store_true',
+                        help='only list remote versions of the package')
+    subparser.add_argument('-n', '--new', action='store_true',
```

Comment:
For the sake of being consistent, can we either use:
```
--safe, --remote, --new
```
or alternatively:
```
--safe-only, --remote-only, --new-only
```
instead of having the `-only` suffix only for the first two options and not for the last?

Reply:
Yes, I can do that. Would prefer the former (less typing) but on the other hand that would change the existing `--safe-only` argument. Any opinions on this?

I didn't add `--new` to the output group initially, because the code logic seemed simpler, as  it still needs to fetch the safe versions to compare. Also you won't get the printout of both safe and new remote versions like this. But the arguments are more consistent with your suggestion, which is always nice.

Reply:
Wait, or did you mean only renaming `--new` to `--new-only` without making it mutually exclusive with `--remote-only`?

Reply:
I vote the shorter flags. If you want to be backwards compatible, you can keep safe-only and add a deprecation warning until the next release. All should be mutually exclusive.

Reply:
@adamjstewart  sounds good - done